packages/nimble-components/src/table-column/text/tests/table-column-text.mdx

```diff
@@ -16,7 +16,13 @@ The <Tag name={tableColumnTextTag}/> column is used to display string fields as
 
 {/* ## Appearance Variants */}
 
-{/* ## Usage */}
+## Usage
+
+### Best Practices
+
+-   Avoid mixing `undefined` and `null` as values for the same field. When grouping this will lead to two groups (one for `null` values and one for `undefined` values) that both have the text `"No value"`.
+-   Avoid mixing empty string with `undefined`/`null`. The distinction when grouping between `"No value"` and `"Empty"` is not likely meaningful to a user.
+-   Avoid displaying whitespace values that are not empty string (`''`) as these values will be rendered as-is in group rows.
```

Comment:
These all essentially boil down to:
-   Use a consistent representation for empty values in your data. Do not mix `null`, `undefined`, and/or the empty string (`''`).
    -   If you want cells to display placeholder text, use `null` or `undefined`. Empty strings are not replaced by placeholder text and will always render as a blank cell. 
    -   Avoid whitespace-only data values, as they can lead to confusing results when grouping or sorting.

This actually makes me wonder why we don't automatically normalize whitespace-only values to the empty string. Is there any reasonable argument not to?

Reply:
I don't think the table should be modifying the data its given, which I think is what you are suggesting. Is that correct?

Reply:
Do we not create a copy of the data passed to `setData` at any point? I wasn't quite sure. If not, and tanstack is sorting/grouping based on those values directly, then I guess we can't really do what I suggested.

Reply:
We do have our own copy of the data, but I don't think the fact we have a copy of the data makes it ok to change the data. Some things that could break if we "magically" change the data are:

- IDs of rows may not be as expected/specified by the client
- Mappings may not work as expected
- Parenting (using IDs) may not work correctly

Granted, those things would only break if people are doing odd things (like having IDs that are whitespace), but it would still be very difficult for a client to debug & understand the behavior if we're essentially using different data than the client gave the table. I think it would be much better for a client to normalize their data rather than the table trying to be too smart about it.